subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaExecSpec.java

```diff
@@ -88,8 +89,8 @@ public void copyTo(JavaExecSpec targetSpec) {
     }
 
     @Override
-    public List<String> getCommandLine() {
-        return argumentsSpec.getCommandLine();
+    public Provider<List<String>> getCommandLine() {
+        return getExecutable().map(executable -> argumentsSpec.getCommandLine());
```

Comment:
💭 This looks a bit funny right now, but currently in JavaExecSpec all properties are not migrated, so commandLine is computed through `argumentsSpec.getCommandLine()`. I plan to improve that with JavaExecSpec migration